tensorflow/compiler/tf2xla/kernels/while_op.cc

```diff
@@ -37,6 +37,30 @@ namespace tensorflow {
 
 namespace {
 
+// Verify that input resources are grouped in the end.
+Status VerifyBodyFunctionSignature(XlaOpKernelContext* ctx,
+                                   const NameAttrList& body_name_attr) {
+  const FunctionBody* body;
+  TF_RETURN_IF_ERROR(ctx->compiler()->FindFunctionBody(body_name_attr, &body));
+  bool has_seen_resource = false;
+  for (int i = 0; i < body->arg_types.size(); i++) {
+    DataType arg_type = body->arg_types[i];
+    if (has_seen_resource) {
+      if (arg_type != DT_RESOURCE) {
+          return errors::InvalidArgument(
+              "Expect input resources are grouped in the end of while body ",
```

Comment:
Let's also use `i` in the error message.  If you think that's not useful then please change the loop to a range for loop.